src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -665,7 +665,7 @@ trait Namers extends MethodSynthesis {
       else assignAndEnterFinishedSymbol(tree)
 
       if (isEnumConstant(tree)) {
-        tree.symbol setInfo ConstantType(Constant(tree.symbol))
+        //tree.symbol setInfo ConstantType(Constant(tree.symbol))
```

Comment:
Also needed to adapt Constructors. It's kind of weird now, it looks something unlinked the fields, as the right tree is displayed in every phase, but when emitting the bytecode `classFields(clasz)` it's not in the decls.